ui/src/components/EntryDetailed/EntryViewer.tsx

```diff
@@ -0,0 +1,83 @@
+import React, {useState} from 'react';
+import styles from './EntryViewer.module.sass';
+import Tabs from "../UI/Tabs";
+import {EntryTableSection, EntryBodySection, EntryTablePolicySection} from "./EntrySections";
+
+const SectionsRepresentation: React.FC<any> = ({data, color}) => {
+    const sections = []
+
+    if (data !== undefined) {
+        for (const [i, row] of data.entries()) {
+            switch (row.type) {
+                case "table":
+                    sections.push(
+                        <EntryTableSection key={i} title={row.title} color={color} arrayToIterate={JSON.parse(row.data)}/>
+                    )
+                    break;
+                case "body":
+                    sections.push(
+                        <EntryBodySection key={i} color={color} content={row.data} encoding={row.encoding} contentType={row.mime_type}/>
+                    )
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    return <>{sections}</>;
+}
+
+const AutoRepresentation: React.FC<any> = ({representation, color, isResponseMocked}) => {
+    const rulesMatched = []
+    const TABS = [
+        {tab: 'request'},
+        {
+            tab: 'response',
+            badge: <>{isResponseMocked && <span className="smallBadge virtual mock">MOCK</span>}</>
+        },
+        {
+            tab: 'Rules',
+        },
+    ];
+    const [currentTab, setCurrentTab] = useState(TABS[0].tab);
+
+    // Don't fail even if `representation` is an empty string
+    if (representation.length === 0) {
```

Comment:
I would do if (representation**?**.length === 0) {

That's way we don't fail even if `representation` is an empty string or null